src/Raven.Client/Documents/Session/DocumentSession.Attachments.cs

```diff
@@ -19,25 +21,71 @@ namespace Raven.Client.Documents.Session
     /// </summary>
     public partial class DocumentSession
     {
-        public AttachmentResult GetAttachment(string documentId, string name, Action<AttachmentResult, Stream> stream)
+        public AttachmentName[] GetAttachmentNames(object entity)
         {
-            var operation = new GetAttachmentOperation(documentId, name, stream, AttachmentType.Document, null);
-            return DocumentStore.Operations.Send(operation);
+            if (entity == null ||
+                DocumentsByEntity.TryGetValue(entity, out DocumentInfo document) == false ||
+                document.Metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments) == false)
+                return Array.Empty<AttachmentName>();
+
+            var results = new AttachmentName[attachments.Length];
+            for (var i = 0; i < attachments.Length; i++)
+            {
+                var attachment = (BlittableJsonReaderObject)attachments[i];
+                results[i] = JsonDeserializationClient.AttachmentName(attachment);
+            }
+            return results;
         }
 
-        public AttachmentResult GetAttachment(object entity, string name, Action<AttachmentResult, Stream> stream)
+        public Stream GetAttachment(string documentId, string name)
```

Comment:
Why not call `GetAttachment(string documentId, string name, out AttachmentDetails attachment)`